zoneinfo/src/tzif.rs

```diff
@@ -0,0 +1,69 @@
+//! Implementation of the `tzif` data struct.
+//!
+//! Please note this currently only supports the minimal required
+//! fields in order to implement a TZif.
+
+// TODO: Potentially add some serialization scheme?
+
+use alloc::{collections::btree_set::BTreeSet, vec, vec::Vec};
+use hashbrown::HashSet;
+
+use crate::{types::Transition, SingleLineZone};
+
+/// A version 2 TZif block.
+///
+/// Please note: implementation is very minimal
+#[derive(Debug)]
+pub struct TzifBlockV2 {
```

Comment:
There are a lot of TZif structs everywhere. I like the convenience of a related struct being in `zoneinfo_compiler`, but I could be convinced that it should live elsewhere.

It might be nice to use structs from `tzif`, but that may require some upstream changes.